homeassistant/components/roborock/config_flow.py

```diff
@@ -137,6 +139,25 @@ async def async_step_code(
             errors=errors,
         )
 
+    async def async_step_dhcp(
+        self, discovery_info: DhcpServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle a flow started by a dhcp discovery."""
+        roborock_entries = self.hass.config_entries.async_entries(DOMAIN)
+        device_registry = dr.async_get(self.hass)
+        for entry in roborock_entries:
+            device_entries = dr.async_entries_for_config_entry(
+                device_registry, config_entry_id=entry.entry_id
+            )
+            for device in device_entries:
+                for connection in device.connections:
+                    if (
+                        connection[0] == "mac"
```

Comment:
how about using the same constant used when the connection is created:
```suggestion
                        connection[0] == dr.CONNECTION_NETWORK_MAC
```